File: server/src/main/java/lighting/server/monitor/MonitorService.java
package lighting.server.monitor;

import org.springframework.messaging.simp.SimpMessagingTemplate;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import lighting.server.artnet.ArtnetListener;

@Component
public class MonitorService {

	private final ArtnetListener artnetListener;
	private final SimpMessagingTemplate messagingTemplate;

	public MonitorService(final SimpMessagingTemplate messagingTemplate, final ArtnetListener artnetListener) {
		this.messagingTemplate = messagingTemplate;
		this.artnetListener = artnetListener;
		this.artnetListener.captureData();
	}

	@Scheduled(fixedDelay = 200)
	public void simulateOutputUpdate() {
		this.messagingTemplate.convertAndSend("/topic/output", artnetListener.getCurrentFrames());
	}
}

File: server/src/main/scala/lassi/server/monitor/MonitorService.scala
package lassi.server.monitor

import lassi.domain.Frame
import lassi.server.artnet.ArtnetListener
import org.apache.logging.log4j.LogManager
import org.springframework.messaging.simp.SimpMessagingTemplate
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

import java.time.LocalDateTime

@Component
class MonitorService(
  messagingTemplate: SimpMessagingTemplate,
  artnetListener: ArtnetListener
) {

  private val log = LogManager.getLogger(classOf[MonitorService])

  @Scheduled(fixedDelay = 200)
  def simulateOutputUpdate(): Unit = {

    val dmxValues = artnetListener.dmxValues(0, 1).map(_.toInt)

    val frame = Frame(
      dmxValues,
      startTime = 0,
      universe = 1,
      createdOn = LocalDateTime.now
    )

    val frames = Map(1 -> frame)

    // log.info("send " + frames)
    this.messagingTemplate.convertAndSend("/topic/output", frames)
  }
}
